packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -127,7 +127,10 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     if (event is PointerMoveEvent) {
       final VelocityTracker tracker = _velocityTrackers[event.pointer];
       assert(tracker != null);
-      tracker.addPosition(event.timeStamp, event.position);
+      // PointerMoveEvent from platform's gesture up events are positionally redundant
+      // vs the last PointerMoveEvent. Do not use in velocity calculation.
+      if (event.sourceEventType != PointerChange.up)
+        tracker.addPosition(event.timeStamp, event.position);
```

Comment:
1. The OS only sends gesture updates every 16ms as far as I can tell. 
2. It seems to always do +/- 10 pixels. Android always gives the exact same positions to the pixel. 
3. touchesMoved 0,0; touchesMoved 1,1; touchesMoved 1,1; touchesEnded 1,1 (if we omit the touchesStarted part). That gets transformed into PointerMoveEvent 0,0; PointerMoveEvent 1,1; PointerMoveEvent 1,1; PointerMoveEvent 1,1 for iOS in the framework.